### PR TITLE
Add Ambient Weather service agent to services/core

### PR DESCRIPTION
**Pull request by jklarson**
# Description

Adds a weather service agent (extends the base weather agent) to VOLTTRON for collecting weather data from Ambient's remote weather station API

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

Includes integration tests using the mock agent framework

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: services/core/Ambient/ambient/agent.py [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+    def get_api_description(self, service_name):
+        """
+        Provides a human-readable description of the various endpoints provided
+        by the agent
+        :param service_name: requested service endpoint
+        :return: Human-readable description string
+        """
+        if service_name is "get_current_weather":
+            "Provides current weather observations by Ambient weather" \
+            " station location name via RPC (Requires " \
```
Comment (schandrika) [inline services/core/Ambient/ambient/agent.py]: Provides current weather observations by Ambient weather station for a give location

File: services/core/Ambient/ambient/agent.py [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+        return isinstance(location.get("location", None), str)
+
+    def get_update_interval(self, service_name):
+        """
+        Indicates the interval between remote API updates
+        :param service_name: requested service endpoint
+        :return: datetime timedelta representing the time interval
+        """
+        if service_name == "get_current_weather":
+            return datetime.timedelta(seconds=3)
```
Comment (schandrika) [inline services/core/Ambient/ambient/agent.py]: is new data really available every 3 seconds?

**Comment by craig8** [comment]
@jklarson please provide the versions of the api and the versions of the ambient weather station that were being tested against.

File: services/core/Ambient/tests/test_ambient_agent.py [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+                                      version, "ambientagent-" + version +
+                                      ".agent-data", "weather.sqlite")
+    sqlite_connection = sqlite3.connect(database_file_path)
+    cursor = sqlite_connection.cursor()
+
+    query_data = query_agent.vip.rpc.call(identity, 'get_current_weather',
+                                          locations).get(timeout=33)
+
+    if query_data[0].get("weather_error"):
+        error = query_data[0].get("weather_error")
```
Comment (bonicim) [inline services/core/Ambient/tests/test_ambient_agent.py:227]: Why is 'error' captured but never used?

File: services/core/Ambient/tests/test_ambient_agent.py [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+    def stop_agent():
+        print("stopping weather service")
+        if volttron_instance.is_running():
+            volttron_instance.stop_agent(agent)
+
+    request.addfinalizer(stop_agent)
+    return agent, identity
+
+
+@pytest.fixture(scope="function")
```
Comment (bonicim) [inline services/core/Ambient/tests/test_ambient_agent.py:175]: FYI, scope [defaults to `function`](https://docs.pytest.org/en/latest/reference.html#pytest-fixture); setting scope to `function` might be unnecessary.

**Pull request merged**